test/integration/targets/cs_vlan_ip_range/tasks/main.yml

```diff
@@ -369,3 +369,67 @@
     that:
     - ipr_net is successful
     - ipr_net is changed
+
+# Create a new zone - the default one is enabled
+- name: assure zone for tests
+  cs_zone:
+    name: cs-test-zone
+    state: present
+    dns1: 8.8.8.8
+    network_type: advanced
+  register: cszone
+
+- name: ensure the zone is disabled
+  cs_zone:
+    name: "{{ cszone.name }}"
+    state: disabled
+
+- name: setup a network for tests
+  cs_physical_network:
+    name: net01
+    zone: "{{ cszone.name }}"
+    isolation_method: VLAN
+    broadcast_domain_range: ZONE
+    state: present
+  register: public_network
+
+- name: setup public network traffic
+  cs_traffic_type:
+    physical_network: "{{ public_network.name }}"
+    traffic_type: Public
+    kvm_networklabel: cloudbr1
+    zone: "{{ public_network.zone }}"
+
+- name: test adding a public IP range
```

Comment:
As I understand from this test, the IP range is created by default to a physical network (which one?) when the 'network' parameter is undefined. Right?
I'm wondering if one of the two parameters should not be specified explicitly to avoid confusion.

Any thoughts?

Reply:
It can but it doesn't have to. This the[ CloudStack code](https://github.com/apache/cloudstack/blob/master/server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java#L3071) that is responsible for this situation:
```java
if (forVirtualNetwork) {
                    // default physical network with public traffic in the zone
                    physicalNetworkId = _networkModel.getDefaultPhysicalNetworkByZoneAndTrafficType(zoneId, TrafficType.Public).getId();
                }
```
If no physical network or virtual network is specified then it will search for a Physical Network with a public traffic type.

In my implementation, I was merely using the fact that this is handled on CloudStack's end.

Reply:
Ok, I'm fine with it. Could you please explicit this behavior in the documentation?

Reply:
Something like this?

```diff
diff --git a/lib/ansible/modules/cloud/cloudstack/cs_vlan_ip_range.py b/lib/ansible/modules/cloud/cloudstack/cs_vlan_ip_range.py
index 24b5f4a70a..967daec440 100644
--- a/lib/ansible/modules/cloud/cloudstack/cs_vlan_ip_range.py
+++ b/lib/ansible/modules/cloud/cloudstack/cs_vlan_ip_range.py
@@ -87,6 +87,7 @@ options:
   for_virtual_network:
     description:
       - true if VLAN is of Virtual type, false if Direct.
+      - If set to C(true) but neither I(physical_network) or I(network) is set CloudStack will try to add the VLAN range to the Physical Network with a Public traffic type.
     type: bool
     default: false
 extends_documentation_fragment: cloudstack
```
(I figured it would be easier to verify this in the discussion than push & launch tests over a documentation line)

Reply:
Good! (perhaps split the line). I just realized that this behavior did not apply to direct vlan creation. If it's the case, I would suggest an additionnal line in the descption of the `network` parameter to indicate that it is required if I(for_virtual_network) and I(physical_network) are not set..

Reply:
Done.